### ISSUE TITLE
Improve accuracy of page detectors

### ISSUE THREAD

**Issue by Witiko** [video699/implementation-system#5]
The low accuracies in [the experimental notebook][accuracies] are worrisome:

Page Detector | Accuracy
---|---
`random` | 0.03%
`siamese` | 3.95%
`imagehash` | 6.58%
`vgg16` | 61.84%
`annotated` | 100%

[The `vgg16` page detector][vgg16] currently uses features produced by the last hiddent layer of a VGG16 model trained on ImageNet. Finetuning may not be an option, since our dataset is ill-suited for classification (too many document pages/classes, too few examples of each class). However, since our dataset is significantly different from ImageNet, we may have better luck using an earlier hidden layer of VGG16:

- [ ] Update [the implementation][vgg16].
- [ ] Optimize [the parameters][vgg16-parameters].
- [ ] [Evaluate][accuracies].

[The `siamese` page detector][siamese] uses [position-dependent samples][siamese-moments] to normalize the input screen images, which may explain why the performance degrades on new document page and screen images ([86% accuracy on training set][siamese-parameters] versus [3.95% accuracy on test set][accuracies]):

- [ ] Update [the implementation][siamese-moments].
- [ ] [Estimate performance via cross-validation][siamese-training].
- [ ] Train and optimize [the parameters][siamese-parameters].
- [ ] [Evaluate the final model][accuracies].

 [vgg16]: https://github.com/video699/implementation-system/blob/master/video699/page/vgg16.py
 [siamese]: https://github.com/video699/implementation-system/blob/master/video699/page/siamese.py
 [siamese-moments]: https://github.com/video699/implementation-system/blob/master/video699/page/siamese.py#L66
 [siamese-parameters]: https://github.com/video699/implementation-system/blob/master/docs/notebooks/page/siamese/parameter_optimization.ipynb
 [siamese-training]: https://github.com/video699/implementation-system/blob/master/docs/notebooks/page/siamese/evaluation.ipynb
 [vgg16-parameters]: https://github.com/video699/implementation-system/blob/master/docs/notebooks/page/vgg16/parameter_optimization.ipynb
 [accuracies]: https://github.com/video699/implementation-system/blob/eff545ad942b9990e5cf5948eab6cd2c1846ecae/docs/notebooks/__main__/speed_and_accuracy.ipynb

**Comment by xbankov** [comment]
- Transfer learning may be a good idea for Siamese.
- Successor to VGG16, e.g. ResNet could be used.
